src/runtime-rs/crates/resource/src/network/network_pair.rs

```diff
@@ -164,6 +164,15 @@ pub async fn create_link(
     Ok(link)
 }
 
+// remove a link by its name
+#[allow(dead_code)]
+pub async fn delete_link(handle: &rtnetlink::Handle, name: &str) -> Result<(), rtnetlink::Error> {
+    let link_index = fetch_index(handle, name)
+        .await
+        .expect("failed to fetch index");
+    handle.link().del(link_index).execute().await
```

Comment:
This code looks racy to me. Could the link index change between the query and the `del().execute()`? If so, we need some locking, but if not, please add some comments about why a lock is not required.

Reply:
And this function is for test only, so it should be in a test mod.

Reply:
Ack. You might be able to use `#[serial]`, but I don't _think_ that works with `async`?

Reply:
@jodh-intel Sorry for the late reply. Thanks for your review! I added a commit for this problem. I think the index(the one that 'ip a' shows) does not change during the link's lifetime, since the index is settled when the link is created.

Reply:
@liubin Fixed now, thanks for the review.